src/GslCore/DnaCreation.fs

```diff
@@ -552,7 +515,28 @@ let expandAssembly
 
                 match ppp.part with
                 | MARKERPART ->
-                    yield expandMarkerPart library dnaSource ppp
+                    // Choose provider
+                    let providers =
+                        markerProviders |> 
+                            List.choose (fun provider -> 
+                                                    match provider.ScoreJob a.capabilities with
+                                                        | None -> None
+                                                        | Some(score) -> Some (score,provider)
+                                        ) 
+                    if providers = [] then failwithf "MARKERPART substitution, no marker provider available"
+                    let _,markerProvider = providers |> List.maxBy (fst)
+
+                    let markerSet =
+                        match a.pragmas.TryGetOne("markerset") with
+                            | Some(x) ->
+                                let x' = x.ToLower()

```

Comment:
Since x is unused after this binding, it would be less error-prone to redefine x here rather than introduce a primed variable.